dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/NettyServiceGateway.scala

```diff
@@ -366,15 +367,22 @@ class NettyServiceGateway(lifecycle: ApplicationLifecycle, config: ServiceGatewa
   }
 
   private val bindFuture = server.bind(config.port).channelFutureToScala
-  lifecycle.addStopHook(() => {
-    for {
-      channel <- bindFuture
-      closed <- channel.close().channelFutureToScala
-    } yield {
-      eventLoop.shutdownGracefully(10, 10, TimeUnit.MILLISECONDS)
-      poolMap.asScala.foreach(_.getValue.close())
+
+  coordinatedShutdown.addTask(
+    CoordinatedShutdown.PhaseServiceUnbind,
+    "unbind-netty-service-gateway"
+  ) {
+      () =>
+        {
+          (for {
+            channel <- bindFuture
+            closed <- channel.close().channelFutureToScala
+          } yield {
+            eventLoop.shutdownGracefully(10, 10, TimeUnit.MILLISECONDS)
```

Comment:
Use `coordinatedShutdown.timeout(CoordinatedShutdown.PhaseServiceUnbind)`?

Reply:
This is what we have in Play NettyServer:

https://github.com/playframework/playframework/blob/61c6f90d7b0c20040d58fbf3e746b7373fdac8fa/framework/src/play-netty-server/src/main/scala/play/core/server/NettyServer.scala#L255-L263

Reply:
After reviewing the Netty implementation and how we use it I think https://github.com/playframework/playframework/blob/61c6f90d7b0c20040d58fbf3e746b7373fdac8fa/framework/src/play-netty-server/src/main/scala/play/core/server/NettyServer.scala#L255-L263 could be better (see the implementation I used here).